README.md

```diff
@@ -222,6 +223,45 @@ comes with significant tradeoffs. As such the current recommendation is:
 function calling it), and document to their users that this function should be called to
 guarantee that probes are registered.
 
+## Notes
+
+The `usdt` crate requires a nightly toolchain, as it relies on the currently-unstable [inline
+asm][inline-asm] feature. However, the crate contains an empty, no-op implementation, which
+generates all the same probe macros, but with empty bodies. This may be selected by passing
+the `--no-default-features` flag when building the crate, or by using `default-features = false`
+in the [`[dependencies]` table][feature-deps] of one's `Cargo.toml`.
+
+Library developers may use `usdt` as an optional dependency, gated by a feature, for example
+named `usdt-probes` or similar. This feature would imply the `usdt/asm` feature, but the `usdt`
+crate could be used with the no-op implementation by default. For example, your `Cargo.toml`
+might contain
+
+```
+[dependencies]
+usdt = { version = "*", optional = true, default-features = false }
+
+# ... later
+
+[features]
+usdt-probes = ["usdt/asm"]
+```
+
+This allows users to opt into probes if they're willing to accept a nightly toolchain.
+
+### The Rust `asm` feature
+
+Recall from the example that the `usdt` crate relies on inline `asm`, which is not yet a
+stable Rust feature. This means that the code _calling_ the generated probe macros must
```

Comment:
in case you want to drop this in there: https://github.com/rust-lang/rust/issues/72016